src/wp-includes/shortcodes.php

```diff
@@ -221,6 +221,17 @@ function do_shortcode( $content, $ignore_html = false ) {
 		return $content;
 	}
 
+	/*
+	 * When wp_get_attachment_image() is called during shortcode rendering,
+	 * we need to make clear that the context is a shortcode and not part
+	 * of the theme's template rendering logic.
+	 */
+	$callback = static function() {
+		return 'do_shortcode';
+	};
+
+	add_filter( 'wp_get_attachment_image_context', $callback );
```

Comment:
Why is this change needed?

Reply:
@felixarntz `has_filter` doesn't work when you use a closure. `_wp_filter_build_unique_id` generates a new unique key every time the closure is used. This only work with string / array callbacks.

Reply:
@spacedmonkey Oh really? I wasn't aware of that. I thought it could still reference the specific closure in some way.

Reply:
@felixarntz Here is a screenshot of me var dump the result of `_wp_filter_build_unique_id`. 

<img width="619" alt="Screenshot 2023-07-10 at 20 21 32" src="https://github.com/WordPress/wordpress-develop/assets/237508/45a2fb6a-0c4c-4980-8d43-e2aa79a34658">
000000007fde4b7b000000001591160c vs 000000007fde4b1e000000001591160c

Reply:
See https://github.com/WordPress/wordpress-develop/pull/4799#discussion_r1258925898, I think the hashes are only different here because we are passing different closures. If we were passing the same closure, the "unique IDs" would also be the same.